File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int maxlen = 0;
        int l = 0;
        int maxfreq = 0;
        vector<int> um(26, 0);
        
        for (int r = 0; r < s.size(); ++r) {
            um[s[r] - 'A']++;
            maxfreq = max(maxfreq, um[s[r] - 'A']);
            
            while (r - l + 1 - maxfreq > k) {
                um[s[l] - 'A']--;
                l++;
                maxfreq = *max_element(um.begin(), um.end());
            }
            
            maxlen = max(maxlen, r - l + 1);
        }
        
        return maxlen;
    }
};

File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.ts
function characterReplacement(s: string, k: number): number {
    let right = 0;
    let left = 0;
    let len = 0;
    let maxFreq = 0;
    let n = s.length;
    let mp = new Map<string, number>();
    while (right < n) {
        let char = s[right];
        mp.set(char, (mp.get(char) || 0) + 1);
        maxFreq = Math.max(maxFreq, mp.get(char));
        if ((right - left + 1) - maxFreq > k) {
            const leftChar = s[left];
            mp.set(leftChar, mp.get(leftChar)! - 1);
            left++;
        }
        len = Math.max(len, right - left + 1);
        right++;

    }
    return len;
};